### ISSUE TITLE
Seems like `preconfigured_waf_config` is causing rules to be recreated every time.

### ISSUE THREAD

**Issue by btkelly** [GoogleCloudPlatform/terraform-google-cloud-armor#94]
This must have occurred in an update as my Cloud Armor config has not changed a lot but now every Terraform apply is causing many rules that have not been changed to be deleted and recreated. When I look at the plan diff the only change I see is the `preconfigured_waf_config` property which is not set and not an option on these rules.

For example, even with the default rule I get the following in a Terraform apply. The only difference I can see is the rule being destroyed has an empty `preconfigured_waf_config` setting which isn't even an option when setting the default rule. But this applies to other types of rules as well I just choose the default as an example.

**Rule being destroyed:**
```
    - rule ***
        - action      = "allow" -> null
        - description = "Default rule, higher priority overrides it" -> null
        - preview     = false -> null
        - priority    = 2147483647 -> null

        - match ***
            - versioned_expr = "SRC_IPS_V1" -> null

            - config ***
                - src_ip_ranges = [
                    - "*",
                  ] -> null
              ***
          ***

        - preconfigured_waf_config ***
          ***
      ***
```

**Replacement rule being created:**
```
    + rule ***
        + action      = "allow"
        + description = "Default rule, higher priority overrides it"
        + preview     = (known after apply)
        + priority    = 2147483647

        + match ***
            + versioned_expr = "SRC_IPS_V1"

            + config ***
                + src_ip_ranges = [
                    + "*",
                  ]
              ***
          ***
      ***
```

**Google Provider Version:** 5.9.0

**Comment by imrannayer** [comment]
@btkelly thats a provider issue. You can recreate this issue by creating policy using resource directly. Can you plz create an issue for the provider [here](https://github.com/hashicorp/terraform-provider-google/issues)

Also see
https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/issues/54

**Comment by mt185252** [comment]
Seems like its happening only with newly created rules. I have a security policy which was created via the GCP console & has a total of 22 rules(again created via the GCP console), one of these rules was added relatively recently when compared to remaining 21.

I am working on terraform import to construct the state of this security policy and I see that after import for this one particular rule, terraform always tries to delete the `preconfigured_waf_config` block & as a result re-creates that rule. All the other 21 are just fine.

If I run `terraform apply` and then immediately run `terraform plan` I see that it again tries to delete  the `preconfigured_waf_config` block.

To confirm my suspicion that this is been happening only with relatively new rules, I performed the below on the mentioned order:

1. Created a new security policy and added 3 rules to it via the GCP console.
2. Imported the state using `terraform import`.
3. Ran `terraform plan` (this showed that the `preconfigured_waf_config` block in the existing rule has to be deleted & hence the rule will be deleted and recreated.)
4. Ran `terraform apply`(It was successful)
5. Immediately ran `terraform plan` again & had same observations as in step 3.

Tried a couple of more times and repeatedly saw this issue occurring.

When I create a policy & rules completely using terraform then I dont see this happening.

**Comment by imrannayer** [comment]
@asdevcapybara this is upstream provider issue. Plz create issue for the provider [here](https://github.com/hashicorp/terraform-provider-google/issues)

**Comment by imrannayer** [comment]
upstream issue: https://github.com/hashicorp/terraform-provider-google/issues/16882

**Comment by imrannayer** [comment]
fixed in 6.11
